running/local-environment.rst

```diff
@@ -0,0 +1,151 @@
+How to setup a local deconst environment
+========================================
+
+Deconst uses homebrew on OSX for dependencies. As such, all of the work in this setup guide is based on a healthy homebrew installation, or installing it fresh.
+
+Update brew database or install homebrew
+----------------------------------------
+
+First, we need to setup (or update) homebrew on your mac. If you *already have homebrew installed*, lets just update your homebrew database\: ::
+
+  $ brew update
+
+If, however, you *don't have homebrew installed*, you can install it with a shell command. You can read more about brew installation on the `homebrew website <http://brew.sh>`_. ::
+
+  $ ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+Once you've either installed or updated your install, you should be able to run :code:`brew --version` and see the current version of your setup. ::
+
+  $ brew --version
+  0.9.5
+
+Install Docker
+--------------
+
+Deconst development is depending on running within docker containers. We'll use homebrew to install docker and various other docker utilities. ::
+
+  $ brew install docker
+  $ brew install docker-compose
+  $ brew install boot2docker
+  $ boot2docker up
+
+Once those steps are completed, we should be able to verify your docker installation is running correctly. ::
+
+  $ docker version
+
+  Client version: 1.6.0
+  Client API version: 1.18
+  Go version (client): go1.4.2
+  Git commit (client): 4749651
+  OS/Arch (client): darwin/amd64
+  Server version: 1.6.0
+  Server API version: 1.18
+  Go version (server): go1.4.2
+  Git commit (server): 4749651
+  OS/Arch (server): linux/amd64
+
+Install Python3
+---------------
+
+Deconst uses python3 for various preparers and other components. We will install this with homebrew as well. ::
+
+  $ brew install python3
+
+  $ python3 --version
+  Python 3.4.3
+
+Install Ruby
+------------
+
+Deconst uses ruby for the jekyll preparer. This is also installed with homebrew. ::
+
+  $ brew install ruby
+
+  $ ruby -v
+  ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
+
+
+Additional Python and Ruby Dependencies
+---------------------------------------
+
+We use **bundler** and **virtualenv** as part of our configuration for invoking components. These are installed with **pip** and **gem** respectively. ::
+
+  $ sudo pip install virtualenv virtualenvwrapper
+  $ sudo gem install bundler
+
+
+Install Node Version Manager (nvm) and Node
+-------------------------------------------
+
+Deconst uses node for the assets preparer. This will be installed via nvm, which allows managing multiple node installations. ::
+
+  $ curl https://raw.githubusercontent.com/creationix/nvm/v0.25.0/install.sh | bash
```

Comment:
This is a hack and should not be necessary. The latest nvm in brew as of today is 0.33.8 so install it from brew.